simdev/module/simdev/similar_dev_searcher.py

```diff
@@ -0,0 +1,147 @@
+        identifier_sentences: List[str] = []
+        for dev_email, repos_info in self.dev_info.items():
+            parts: List[str] = []
+            for repo, repo_info in repos_info.items():
+                unrolled_identifiers = [
+                    identifier
+                    for (identifier, times) in repo_info['identifiers'].items()
+                    for _ in range(times)
+                ]
+                parts.append(' '.join(unrolled_identifiers))
+            identifier_sentences.append(' '.join(parts))
+
+        identifier_vectors = vectorizer.fit_transform(identifier_sentences).todense()
+        identifier_feature_names = vectorizer.get_feature_names_out()
+        return identifier_vectors, identifier_feature_names
+
+    def _vectorize_langs(self) -> Tuple[np.matrix, np.ndarray]:
+        """
+        Collect languages and form a matrix out of them for all the developers
+        using DictVectorizer
+        :return: Tuple of: vectors and feature (language) names
+        """
+        vectorizer = DictVectorizer()
+        lang_dict: Dict[str, Counter] = defaultdict(Counter)
+        for dev_email, repos_info in self.dev_info.items():
+            for repo, repo_info in repos_info.items():
+                lang_dict[dev_email].update(**repo_info['langs'])
+
+        lang_vectors = vectorizer.fit_transform(lang_dict.values()).todense()
+        lang_feature_names = vectorizer.get_feature_names_out()
+        return lang_vectors, lang_feature_names
+
+    def _compute_sorted_scores(self,
+                               df: pd.DataFrame,
+                               dev_email: str) -> Dict[str, float]:
+        """
+        Compute sorted similarity scores using cosine similarity
+        :param df: Prepared dataframe
+        :param dev_email: Email of the developer to find similar to
+        :return: Dict from developer's email to its score.
+        Items are sorted by value (The highest score - first key)
+        """
+        # Compute scores filtering dataframe
+        scores = cosine_similarity(df[df.index != dev_email], df[df.index == dev_email])
+        # "Join" emails and scores, sort it by score and convert to dict
+        email_score_array = np.column_stack((df[df.index != dev_email].index, scores))
+        sorted_scores: Dict[str, float] = \
+            dict(sorted(email_score_array,
+                        key=lambda row: row[1],
+                        reverse=True)[:self.max_results_count])
+        return sorted_scores
+
+    def _write_top_params(self, dev_email: str, result: SimilarDevInfo) -> None:
+        """
+        Write most frequently used params: languages, identifiers
+        and repositories (based on the file count modified in them)
+        :param dev_email: dev email to write info about
+        :param result: to write top params to
+        :return: dict of most common identifiers
+        """
+        identifiers_counter = Counter({})
```

Comment:
```suggestion
        identifiers_counter = Counter()
```
you don't need to specify dictionary - it will be initialized on empty dictionary - so it's the same as creation of `Counter`